pkg/controller/common/unmanaged.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package common
+
+import (
+	"strconv"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	// ManagedAnnotation annotation
+	LegacyPauseAnnoation = "common.k8s.elastic.co/pause"
+	ManagedAnnotation    = "eck.k8s.elastic.co/managed"
+)
+
+var (
+
+	// CheckManagedRequeue is the default requeue to re-check if a resource is still unmanaged.
+	CheckManagedRequeue = reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}
+)
+
+// IsUnmanaged checks if a given resource is currently unmanaged.
+func IsUnmanaged(meta metav1.ObjectMeta) bool {
+	return getBoolFromAnnotation(meta.Annotations)
+}
+
+// Extract the desired state from the map that contains annotations.
+func getBoolFromAnnotation(annotations map[string]string) bool {
+	if annotations == nil {
+		return false
+	}
+
+	test := func(ann string, defaultValue bool) bool {
+		stateAsString, exists := annotations[ann]
+		if !exists {
+			return defaultValue
+		}
+
+		expectedState, err := strconv.ParseBool(stateAsString)
+		if err != nil {
+			log.Error(err, "Cannot parse %s as a bool, defaulting to %s: \"%v\"", annotations[ann], ann, defaultValue)
+			return defaultValue
+		}
+		return expectedState
+	}
+	return test(LegacyPauseAnnoation, false) || !test(ManagedAnnotation, true)
```

Comment:
What should happen if both annotations are present? Here we are ignoring the `managed` annotation if the `pause` annotation exists. Should it be the other way around?

Reply:
I was not sure myself. I was mainly thinking about not disrupting any existing clusters with the `pause` annotation so I gave that preference. But I can see your point, and I guess it would not hurt to prefer the new annotation.

Reply:
Maybe we can log which annotation was used to make the decision? That way if users are inadvertently using both annotations, they could at least see that only one of them is having an effect.

Reply:
OK I have changed it to 
1. prefer the new annotation
2. log if a user is still using the old one and if the user uses both. 

Logic is a bit convoluted but hopefully clearer for the user.